dask/dataframe/shuffle.py

```diff
@@ -370,7 +343,12 @@ def set_partition(
             column_dtype=df.columns.dtype,
         )
 
-    df4.divisions = tuple(methods.tolist(divisions))
+    # None and pd.NA values are not sortable
+    divisions = methods.tolist(divisions)
+    if pd.isna(divisions).all():
```

Comment:
Do we run into problems if divisions has some, but not all, `pd.NA` / `None` entries?

Reply:
Yes, I think so. I just haven't seen a test case like that.